test/support.ts

```diff
@@ -0,0 +1,350 @@
+}
+
+export function getSequelizeInstance(db: string, user: string, pass: string, options?: Options) {
+  options = options || {};
+  options.dialect = options.dialect || getTestDialect();
+
+  return new Sequelize(db, user, pass, options);
+}
+
+export async function clearDatabase(sequelize: Sequelize) {
+  const qi = sequelize.getQueryInterface();
+  await qi.dropAllTables();
+  sequelize.modelManager.models = [];
+  sequelize.models = {};
+
+  if (qi.dropAllEnums) {
+    await qi.dropAllEnums();
+  }
+
+  await dropTestSchemas(sequelize);
+}
+
+export async function dropTestSchemas(sequelize: Sequelize) {
+  const queryInterface = sequelize.getQueryInterface();
+
+  if (!queryInterface.queryGenerator._dialect.supports.schemas) {
+    await sequelize.drop({});
+
+    return;
+  }
+
+  const schemas = await sequelize.showAllSchemas();
+  const schemasPromise = [];
+  for (const schema of schemas) {
+    // @ts-expect-error
+    const schemaName = schema.name ? schema.name : schema;
+    if (schemaName !== sequelize.config.database) {
+      schemasPromise.push(sequelize.dropSchema(schemaName));
+    }
+  }
+
+  await Promise.all(schemasPromise.map(async p => p.catch((error: unknown) => error)));
+}
+
+export function getSupportedDialects() {
+  return fs.readdirSync(path.join(distDir, 'dialects'))
+    .filter(file => !file.includes('.js') && !file.includes('abstract'));
+}
+
+// TODO: type once QueryGenerator has been migrated to TS
+export function getAbstractQueryGenerator(sequelize: Sequelize): unknown {
+  class ModdedQueryGenerator extends AbstractQueryGenerator {
+    quoteIdentifier(x: string): string {
+      return x;
+    }
+  }
+
+  // @ts-expect-error
+  return new ModdedQueryGenerator({ sequelize, _dialect: sequelize.dialect });
+}
+
+export function getTestDialect(): Dialect {
+  let envDialect = process.env.DIALECT || 'mysql';
+
+  if (envDialect === 'postgres-native') {
+    envDialect = 'postgres';
+  }
+
+  if (!getSupportedDialects().includes(envDialect)) {
+    throw new Error(`The dialect you have passed is unknown. Did you really mean: ${envDialect}`);
+  }
+
+  return envDialect as Dialect;
+}
+
+export function getTestDialectTeaser(moduleName: string): string {
+  let dialect: string = getTestDialect();
+
+  if (process.env.DIALECT === 'postgres-native') {
+    dialect = 'postgres-native';
+  }
+
+  return `[${dialect.toUpperCase()}] ${moduleName}`;
+}
+
+export function getPoolMax(): number {
+  return Config[getTestDialect()].pool?.max ?? 1;
+}
+
+type ExpectationKey = Dialect | 'default';
+type PartialRecord<K extends keyof any, V> = Partial<Record<K, V>>;
+
+export function expectsql(
+  query: { query: string, bind: unknown } | Error,
+  assertions: { query: PartialRecord<ExpectationKey, string | Error>, bind: PartialRecord<ExpectationKey, unknown> },
+): void;
+export function expectsql(
+  query: string | Error,
+  assertions: PartialRecord<ExpectationKey, string | Error>,
+): void;
+export function expectsql(
```

Comment:
Why are we exporting this multiple times?

Reply:
It's overloading (https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads)

Reply:
Ah yes, I was looking for the term of this already. But the thing I am missing here is why we use this here, because we have 2 arguments in all three options. And the third just looks like a merger of the first two. But probably I'm missing something here in my thought process

Reply:
The implementation is always a merger of all previous overloads, but it's 'internal' of sorts (even if exported), TypeScript won't include it as part of the possible overloads when you use the method